File: frontend-server/src/app/model/dto/summation-dto.ts
import { Employee } from "../employee";
import { BasicDto } from "./basic-dto";

export class SummationDto extends BasicDto<Employee, SummationResult> {
  from:       string;
  to:         string;
  positionId: number;
  shiftId:    number;
}

export class SummationResult {
  summationColumnId:  number;
  type:               string;
  value:              number;
}

export enum SummationMode {
  OVERALL       = "overall",
  PER_POSITION  = "per_position"
}

File: core/src/main/java/com/averude/uksatse/scheduler/core/model/dto/SummationDTO.java
package com.averude.uksatse.scheduler.core.model.dto;

import com.averude.uksatse.scheduler.core.interfaces.entity.HasPositionId;
import com.averude.uksatse.scheduler.core.interfaces.entity.HasShiftId;
import com.averude.uksatse.scheduler.core.model.entity.Employee;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.io.Serializable;
import java.time.LocalDate;
import java.util.List;

@Getter
@Setter
@NoArgsConstructor
public class SummationDTO extends BasicDto<Employee, SummationResult> implements Serializable, HasPositionId, HasShiftId {
    private LocalDate from;
    private LocalDate to;
    private Long positionId;
    private Long shiftId;

    public SummationDTO(Employee parent,
                        LocalDate from,
                        LocalDate to,
                        List<SummationResult> collection) {
        super(parent, collection);
        this.from = from;
        this.to = to;
    }
}
